pynif/tests/test_context.py

```diff
@@ -20,7 +22,51 @@ def setUpClass(cls):
                 nif:endIndex    "{}"^^xsd:nonNegativeInteger ;
                 nif:isString    "{}" .
         """.format(len(cls.example_text), cls.example_text.replace('"', '\\"'))
-    
+        cls.example_turtle_ContextHashBasedString = """
+            @prefix xsd:   <http://www.w3.org/2001/XMLSchema#> .
+            @prefix itsrdf: <http://www.w3.org/2005/11/its/rdf#> .
+            @prefix nif:   <http://persistence.uni-leipzig.org/nlp2rdf/ontologies/nif-core#> .
+                    
+            <http://www.cse.iitb.ac.in/~soumen/doc/CSAW/doc#hash_0_1411_6218664a3a8c7bed58460e329ddc6904_%20%20%20%20Primary%20Navigati>
+                a                       nif:ContextHashBasedString , nif:Context ;
+                nif:beginIndex  "0"^^xsd:nonNegativeInteger ;
+                nif:endIndex    "{}"^^xsd:nonNegativeInteger ;
+                nif:isString    "{}" .
+        """.format(len(cls.example_text), cls.example_text.replace('"', '\\"'))
+        # The following string is copied from the file tests/data/example-ContextHashBasedString.ttl
+        # The load from file would not pass GitHub Checks sinse the file example-ContextHashBasedString.ttl 
+        # it is not in the main repo. Once the PR is completed the  line 
+        # cls.example_ContextHashBasedString = """....""" can be replaced by the following code:
```

Comment:
I might have misunderstood something, but why cannot you commit the file to your branch? This should make it available in the tests and would add it to the main repo once we merge the PR.

Reply:
For some reason if I add these lines to the test_collection, the GitHub Check gives an error. It tells that the file cannot be found and the file it is in my repo (I added a new file at tests/data).
        #testdir = os.path.dirname(os.path.abspath(__file__))
        #with open(os.path.join(testdir, 'data/example-ContextHashBasedString.ttl'), 'r') as f:
        #   cls.example_ContextHashBasedString = f.read()

Reply:
This is the error:
        testdir = os.path.dirname(os.path.abspath(__file__))
>       with open(os.path.join(testdir, 'data/example-ContextHashBasedString.ttl'), 'r') as f:
E       FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/pynif/pynif/pynif/tests/data/example-ContextHashBasedString.ttl'

Reply:
I think you just haven't added it to your commits. If you are using the command-line, `git add tests/data/example-ContextHashBasedString.ttl`, `git commit` and `git push` should do. If you are using some GUI, then there should be an option to add untracked files.

Reply:
Solved! It worked now. It was a case sensitive problem. ``data/example-ContextHashBasedString.ttl`` has a capital C and it is not. I just changed to ``data/example-contextHashBasedString.ttl`` and it work